src/config/mod.rs

```diff
@@ -164,13 +223,20 @@ pub struct SourceOuter {
     pub proxy: ProxyConfig,
     #[serde(flatten)]
     pub(super) inner: Box<dyn SourceConfig>,
+    #[serde(
+        default,
+        skip_deserializing,
+        skip_serializing_if = "vector_core::serde::skip_serializing_if_default"
```

Comment:
We should always skip serializing, no?

Reply:
I guess maybe this shouldn't even be annotated as `serde`? It should never be read from the configuration, but rather set by sinks?

Reply:
The trick is that the sink acknowledgements are propagated at the config level, while the `SourceContext` is only created on-the-fly as sources are built. I _think_ the only place I can store this data is at the `SourceOuter` level. I have rewritten this to `#[serde(skip)]` which means serde just ignores it and resets it to default.